src/System.Private.Xml/src/System/Xml/Core/XmlEncodedRawTextWriter.cs

```diff
@@ -1077,7 +1077,7 @@ protected unsafe void WriteElementTextBlock(char* pSrc, char* pSrcEnd)
                             {
                                 case NewLineHandling.Replace:
                                     // Replace "\r\n", or "\r" with NewLineChars
-                                    if (pSrc[1] == '\n')
+                                    if (&pSrc[1] < pSrcEnd && pSrc[1] == '\n')
```

Comment:
Instead of using `&pSrc[1] < pSrcEnd`, could you use `pSrc + 1 < pSrcEnd` or `pSrc < pSrcEnd - 1`? My reasoning is:
a) That's the style used elsewhere in the file, and generally throughout corefx.
b) Even though it's not actually dereferencing, it looks like it is, and so isn't as easy to review.
c) The latter will actually generate slightly more concise IL.